oak_channel/SPEC.md

```diff
@@ -44,91 +43,47 @@ The service MUST NOT send request messages.
 ##### Handling an invocation
 
 The service MUST respond to each request message with a single response message.
-It SHOULD respond to request messages in first-in-first-out order, but MAY
-prioritize some request messages. The service MAY send a response message before
-the request message has been received in its entirety. Sending a response
-message concludes the invocation.
+The service MAY send a response message before the request message has been
+received in its entirety, but MUST NOT send start sending a response before any
+part of the request message has been received. Sending a response message
+concludes the invocation.
 
 The client MUST NOT send response messages.
 
 ### Message Layer
 
-Messages are conceptual units to transport requests and responses.
+Messages are logical units used to transport requests and responses.
 
 There are exactly two messages types: request messages and response messages.
+Since the invocation layer is responsible for encoding requests and reponses, at
+the communication protocol level the encoding for request and response messages
+follow an identical format.
```

Comment:
Worth mentioning that requests will always "flow" in one direction, and responses in the opposite direction? I.e. that each logical channel has fixed roles for client and server, and invocations are always initiated by the client.